### PR TITLE
ASC-350 Update proxy tests with more applications operations

### PR DESCRIPTION
**Pull request by jkopriva**

### CODE REVIEW

File: test/e2e/proxy_test.go
Comment (alexeykazakov) [inline test/e2e/proxy_test.go:218]: Instead of re-creating all this infrastructure for every sub test (proxy client, websocket watcher, etc) I would just encapsulate the sub-tests to reproduce the whole flow:

```
t.Run("use proxy to create a HAS Application CR in the user appstudio namespace via proxy API and use websocket to watch it created", func(t *testing.T) {
    // Create a proxy client, start a watcher, etc
    // Create an app, watch it created, check you can get it via proxy
    ...
    t.Run("use proxy to update a HAS Application...", func(t *testing.T) {
        // Reuse the proxy client, watcher, etc created above
        // Update the app
        ...
       t.Run("use proxy to delete a HAS Application...", func(t *testing.T) {
            // Reuse the proxy client, watcher, etc created above
            // Delete the app
           ...
        }
    }
}
```

Or alternatively:
```
// Create a proxy client, start a watcher, etc
...
t.Run("use proxy to create a HAS Application CR in the user appstudio namespace via proxy API and use websocket to watch it created", func(t *testing.T) {
    // Reuse the proxy client, watcher, etc created above
    // Create an app, watch it created, check you can get it via proxy
    ...
}

t.Run("use proxy to update the HAS Application created above...", func(t *testing.T) {
    // Reuse the proxy client, watcher, etc created above
    // Update the app
    ...
}

t.Run("use proxy to delete the HAS Application created above...", func(t *testing.T) {
    // Reuse the proxy client, watcher, etc created above
    // Delete the app
    ...
}
```

Also, I think we don't have to retry every operation multiple times to test the cache. It's enough to do it just when we are creating the app in the very first test.

**Review by alexeykazakov — COMMENTED** [review]
Looks good! I have a few comments to address though.

File: test/e2e/proxy_test.go
Comment (jkopriva) [reply test/e2e/proxy_test.go]: I have deleted this test and moved user.getApplication(...) to create application test. Resolving.

File: test/e2e/proxy_test.go
Comment (jkopriva) [reply test/e2e/proxy_test.go]: I have rewritten code to use the second approach. Resolving.

**Comment by jkopriva** [comment]
@alexeykazakov Hi, thank you for review and comments. I have resolved all issues, can you please review this PR again? Thank you!

File: test/e2e/proxy_test.go
Comment (alexeykazakov) [inline test/e2e/proxy_test.go:226]: And I don't see any websocket usage here either.

File: test/e2e/proxy_test.go
Comment (alexeykazakov) [inline test/e2e/proxy_test.go:256]: Or you could use the application list you got from the test above to delete them all.

File: test/e2e/proxy_test.go
Comment (alexeykazakov) [inline test/e2e/proxy_test.go:273]: Check the error to make sure it's actually 404.

**Review by alexeykazakov — COMMENTED** [review]
Looks better!

File: test/e2e/proxy_test.go
Comment (jkopriva) [reply test/e2e/proxy_test.go]: Thank you! Added.

File: test/e2e/proxy_test.go
Comment (jkopriva) [reply test/e2e/proxy_test.go]: Rewritten to use application list from previous test.

File: test/e2e/proxy_test.go
Comment (jkopriva) [reply test/e2e/proxy_test.go:272]: Added check for k8serr.IsNotFound(err).

File: test/e2e/proxy_test.go
Comment (jkopriva) [reply test/e2e/proxy_test.go]: Updated description.

**Review by alexeykazakov — APPROVED** [review]
Thank you for addressing my comments.

File: test/e2e/proxy_test.go
Comment (MatousJobanek) [inline test/e2e/proxy_test.go:192]: this test (and also the following ones) depends on the execution of the first one, so they should be implemented as sub-tests of the first one, and not at the same level.

File: test/e2e/proxy_test.go
Comment (MatousJobanek) [inline test/e2e/proxy_test.go:936]: `present` is already a boolean
```suggestion
		return !present, nil
```

File: test/e2e/proxy_test.go
Comment (MatousJobanek) [inline test/e2e/proxy_test.go:263]: I know that this was already there before, but we could use the "default constants" for these values inside of the `WaitFor*` methods instead of passing it every time.
https://github.com/codeready-toolchain/toolchain-e2e/blob/2b54e59598f40d93c94c2a457d5ebe96cca4f962/testsupport/wait/awaitility.go#L38-L39

**Review by MatousJobanek — COMMENTED** [review]
Cool, thanks a lot for your first contribution :+1: 
I left a few comments. One more question. I guess that you are going to add additional tests in the following PRs and later introduce also parallelization of these tests, is that correct?

**Comment by jkopriva** [comment]
Hi @MatousJobanek, thank you for all comments and suggestions! I will fix the issues, wait until this one is merged and open new PR with components and so on.

File: test/e2e/proxy_test.go
Comment (jkopriva) [reply test/e2e/proxy_test.go]: Resolved. Thank you!

File: test/e2e/proxy_test.go
Comment (jkopriva) [reply test/e2e/proxy_test.go]: Resolved. Thank you!

File: test/e2e/proxy_test.go
Comment (jkopriva) [reply test/e2e/proxy_test.go]: Resolved. Thank you!

File: test/e2e/proxy_test.go
Comment (jkopriva) [reply test/e2e/proxy_test.go]: Resolved. Thank you!

File: test/e2e/proxy_test.go
Comment (jkopriva) [reply test/e2e/proxy_test.go]: Resolved. Thank you!

File: test/e2e/proxy_test.go
Comment (jkopriva) [reply test/e2e/proxy_test.go]: Resolved. Thank you!

File: test/e2e/proxy_test.go
Comment (jkopriva) [reply test/e2e/proxy_test.go]: Good point. It should be resolved now. Thank you!

File: test/e2e/proxy_test.go
Comment (jkopriva) [reply test/e2e/proxy_test.go]: Resolved. Thank you!

**Comment by jkopriva** [comment]
@MatousJobanek Hi, when you have time, PTAL, all issues should be resolved. Thank you!

**Review by MatousJobanek — APPROVED** [review]
Thanks a lot for the additional changes. 
I left a few suggestions for better variable naming and better handling of the assertions, otherwise, it looks good :+1: :1st_place_medal: :rocket:

File: test/e2e/proxy_test.go
Comment (jkopriva) [reply test/e2e/proxy_test.go]: Thank you! Resolved.

File: test/e2e/proxy_test.go
Comment (jkopriva) [reply test/e2e/proxy_test.go:168]: ok. WiIl do in next PRs :)

**Pull request merged**
